sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateUnsafeProjection.scala

```diff
@@ -316,8 +316,10 @@ object GenerateUnsafeProjection extends CodeGenerator[Seq[Expression], UnsafePro
   protected def canonicalize(in: Seq[Expression]): Seq[Expression] =
     in.map(ExpressionCanonicalizer.execute)
 
-  protected def bind(in: Seq[Expression], inputSchema: Seq[Attribute]): Seq[Expression] =
-    in.map(BindReferences.bindReference(_, inputSchema))
+  protected def bind(in: Seq[Expression], inputSchema: Seq[Attribute]): Seq[Expression] = {
+    lazy val inputSchemaAttrSeq: AttributeSeq = inputSchema
```

Comment:
Why the lazy val? Are you optimizing for the case where in is empty?

Reply:
Yes, that is the reason. For example, the query <code>df.count</code>, where df is a dataframe from a CSV datasource, calls GenerateUnsafeProjection.bind with am empty list of expressions.

However, the map inside the AttributeSeq object is not built until someone accesses exprIdToOrdinal, so maybe it is overkill.

Reply:
what about changing the signature of `bind` instead?

Reply:
@mgaido91 Do you mean change it to?

```scala
bind(in: Seq[Expression], inputSchema: AttributeSeq): Seq[Expression]
```

Reply:
yes

Reply:
+1 on something that eliminates this issue wholesale.